docs/single-to-multi-master.md

```diff
@@ -0,0 +1,213 @@
+
+```bash
+root@ip-172-20-116-230:~# systemctl start protokube
+```
+
+## 3 - Add the third master
+
+### a - Create the instance group
+
+Create 1 kops instance group for the third master, in
+a different AZ from the existing ones.
+
+```bash
+$ kops create instancegroup master-<availability-zone3>
+```
+
+ * ``maxSize`` and ``minSize`` should be 1,
+ * ``role`` should be ``Master``,
+ * only one zone should be listed.
+
+### b - Add a new member to the etcd clusters
+
+ ```bash
+ $ kubectl --namespace=kube-system exec etcd-server-ip-172-20-36-161.ec2.internal -- etcdctl member add etcd-<availability-zone3> http://etcd-<availability-zone3>.internal.myclusterdomain.com:2380
+ $ kubectl --namespace=kube-system exec etcd-server-events-ip-172-20-36-161.ec2.internal -- etcdctl --endpoint http://127.0.0.1:4002 member add etcd-events-<availability-zone3> http://etcd-events-<availability-zone3>.internal.myclusterdomain.com:2381
+ ```
+
+### c - Launch the third master
+
+ ```bash
+ $ kops update cluster myclusterdomain.com --yes
+ # wait for the third master to boot and initialize
+ $ ssh admin@<third-master>
+ admin@ip-172-20-139-130:~$ sudo -i
+ root@ip-172-20-139-130:~# systemctl stop kubelet
+ root@ip-172-20-139-130:~# systemctl stop protokube
+ ```
+
+ Reinitialize the etcd instances:
+ * In both ``/etc/kubernetes/manifests/etcd-events.manifest`` and
+ ``/etc/kubernetes/manifests/etcd.manifest``, edit the
+ ``ETCD_INITIAL_CLUSTER_STATE`` variable to ``existing``.
+ * Delete the containers and the data directories:
+
+ ```bash
+ root@ip-172-20-139-130:~# docker stop $(docker ps | grep "etcd:2.2.1" | awk '{print $1}')
+ root@ip-172-20-139-130:~# rm -r /mnt/master-vol-019796c3511a91b4f//var/etcd/data-events/member/
+ root@ip-172-20-139-130:~# rm -r /mnt/master-vol-0c89fd6f6a256b686/var/etcd/data/member/
+ ```
+
+ Launch them again:
+
+ ```bash
+ root@ip-172-20-139-130:~# systemctl start kubelet
+ ```
+
+ At this point, both etcd clusters should be healthy with three members:
+
+ ```bash
+ $ kubectl --namespace=kube-system exec etcd-server-ip-172-20-36-161.ec2.internal -- etcdctl member list
+ $ kubectl --namespace=kube-system exec etcd-server-ip-172-20-36-161.ec2.internal -- etcdctl cluster-health
+ $ kubectl --namespace=kube-system exec etcd-server-events-ip-172-20-36-161.ec2.internal -- etcdctl --endpoint http://127.0.0.1:4002 member list
+ $ kubectl --namespace=kube-system exec etcd-server-events-ip-172-20-36-161.ec2.internal -- etcdctl --endpoint http://127.0.0.1:4002 cluster-health
+ ```
+
+ If not, check ``/var/log/etcd.log`` for problems.
+
+ Restart protokube on the third master:
+
+ ```bash
+ root@ip-172-20-139-130:~# systemctl start protokube
+ ```
+
+## 4 - Cleanup
+
+To be sure that everything runs smoothly and is setup correctly, it is advised
+to terminate the masters one after the other (always keeping 2 of them up and
+running). They will be restarted with a clean config and should join the others
+without any problems.
```

Comment:
I included this step for 2 reasons:

- to be sure that the masters we end up with are fully configured by kops and there is no residual manual configuration,
- to catch any configuration problem right now instead of later, at the next update or failure of a master.

I can include those reasons in the doc or simply delete this paragraph. Which do you prefer?